ci/run.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+# Builds and runs tests for a particular target passed as an argument to this
+# script.
+
+set -e
+
+BASE_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+MANIFEST_PATH="${BASE_DIR}/Cargo.toml"
+BUILD_DIR="."
+
+VERSION="$1"
+TARGET="$2"
+
+export RUST_TEST_THREADS=1
+
+echo "======================================================="
+echo "TESTING VERSION: ${VERSION}, TARGET: ${TARGET}"
+echo "======================================================="
+
+#
+# Tell cargo what linker to use and whatever else is required
+#
+configure_cargo() {
+  rm -rf .cargo
```

Comment:
eg, I'd rather save the existing one, and have a trap to put the original back on exit.